### PR TITLE
Implement vector mode: intrinsics

### PR DESCRIPTION
**Pull request by tgymnich**

### CODE REVIEW

File: enzyme/Enzyme/AdjointGenerator.h [hunk 1]
```diff
@@ -3427,16 +3445,23 @@ class AdjointGenerator
...
+                            ? Constant::getNullValue(opType1)
                             : diffe(orig_ops[1], Builder2);
 
-        Value *dif = Builder2.CreateSelect(cmp, diffe0, diffe1);
-        setDiffe(&I, dif, Builder2);
+        auto rule = [&](Value *diffe0, Value *diffe1) {
+          return Builder2.CreateSelect(cmp, diffe0, diffe1);
+        };
 
+        Value *dif =
```
Comment (wsmoses) [inline enzyme/Enzyme/AdjointGenerator.h:3462]: Does this work still if only one of diffe0 is vectorized/active, and the other is not?

Comment (wsmoses) [reply enzyme/Enzyme/AdjointGenerator.h:3462]: I would've expected for ease, you to use the diffe0 = getShadowType always, so then both are vectorized

File: enzyme/test/Enzyme/ForwardModeVector/maxnum.ll [hunk 2]
```diff
@@ -0,0 +1,37 @@
+; RUN: %opt < %s %loadEnzyme -enzyme -enzyme-preopt=false -mem2reg -early-cse -simplifycfg -S | FileCheck %s
+
+%struct.Gradients = type { double, double }
+
+; Function Attrs: nounwind
+declare %struct.Gradients @__enzyme_fwddiff(double (double, double)*, ...)
```
Comment (wsmoses) [inline enzyme/test/Enzyme/ForwardModeVector/maxnum.ll:6]: For the multi-arg tests, can you add versions where only one argument is active?

File: enzyme/Enzyme/AdjointGenerator.h (hunk 1 above)
Comment (tgymnich) [reply enzyme/Enzyme/AdjointGenerator.h:3462]: fixed

File: enzyme/test/Enzyme/ForwardModeVector/maxnum.ll (hunk 2 above)
Comment (tgymnich) [reply enzyme/test/Enzyme/ForwardModeVector/maxnum.ll:6]: done

**Review by wsmoses — APPROVED** [review]
Fix the LLVM test otherwise LGTM

**Pull request merged**